src/box/lua/serialize_lua.c

```diff
@@ -955,7 +955,7 @@ dump_root(struct lua_dumper *d)
 
 	luaL_checkfield(d->L, d->cfg, lua_gettop(d->L), &nd.field);
 
-	if (nd.field.type != MP_ARRAY || nd.field.size != 1) {
+	if (nd.field.type != MP_ARRAY || nd.field.size > 1) {
```

Comment:
I can't reproduce the error from the commit description, either with or without this patch.

Reply:
Please try master + this patch: https://github.com/tarantool/tarantool/pull/7085/commits/da35fe3de24ddc0c4a4dc75ace782d3c02e28cdd